commcare_connect/opportunity/forms.py

```diff
@@ -1286,106 +1288,165 @@ def __init__(self, *args, **kwargs):
+        return waffle.switch_is_active("automated_invoices")
+
+    def invoice_form_fields(self):
+        invoice_number_attrs = {}
+        if self.is_automated_invoice:
+            invoice_number_attrs["readonly"] = "readonly"
+        first_row = [Field("invoice_number", **invoice_number_attrs)]
+
+        if self.is_service_delivery:
+            first_row.append(Field("title"))
+
+        second_row = [Field("date", **{"x-ref": "date", "x-on:change": "convert()"})]
+        if self.is_service_delivery:
+            second_row.append(Field("start_date"))
+            second_row.append(Field("end_date"))
+
+        return [
+            Div(
+                Div(*first_row, css_class="grid grid-cols-3 gap-6"),
+                Div(*second_row, css_class="grid grid-cols-3 gap-6"),
+                Div(
+                    Field(
+                        "local_amount",
+                        label=f"Amount ({self.opportunity.currency})",
+                        **{
+                            "x-ref": "amount",
+                            "x-on:input.debounce.300ms": "convert()",
+                        },
+                    ),
+                    Field("amount_usd"),
+                    Div(css_id="converted-amount-wrapper", css_class="space-y-1 text-sm text-gray-500 mb-4"),
+                    css_class="grid grid-cols-3 gap-6",
+                ),
+                css_class="flex flex-col gap-4",
+            ),
+        ]
+
+    def generate_invoice_number(self):
+        return uuid.uuid4().hex[:10].upper()
+
     def clean_invoice_number(self):
         invoice_number = self.cleaned_data["invoice_number"]
-        if PaymentInvoice.objects.filter(opportunity=self.opportunity, invoice_number=invoice_number).exists():
+
+        if not invoice_number:
+            invoice_number = self.generate_invoice_number()
+
+        if PaymentInvoice.objects.filter(invoice_number=invoice_number).exists():
             raise ValidationError(
-                f'Invoice "{invoice_number}" already exists',
+                "Please use a different invoice number",
                 code="invoice_number_reused",
             )
         return invoice_number
 
     def clean(self):
         cleaned_data = super().clean()
-        amount = cleaned_data.get("amount")
-        usd_currency = cleaned_data.get("usd_currency")
+        local_amount = cleaned_data.get("local_amount")
         date = cleaned_data.get("date")
 
-        if amount is None or date is None:
+        if local_amount is None or date is None:
             return cleaned_data  # Let individual field errors handle missing values
 
         exchange_rate = ExchangeRate.latest_exchange_rate(self.opportunity.currency, date)
         if not exchange_rate:
             raise ValidationError("Exchange rate not available for selected date.")
 
         cleaned_data["exchange_rate"] = exchange_rate
+        cleaned_data["amount_usd"] = round(local_amount / exchange_rate.rate, 2)
```

Comment:
same comment as the other one about exchange rates. if we arent enforcing these to be single month *cough* its not too late *cough*, this logic won't be accurate since each item might have a different exchange rate.